riscv/src/compiler.rs

```diff
@@ -214,44 +215,62 @@ fn next_multiple_of_four(x: usize) -> usize {
     ((x + 3) / 4) * 4
 }
 
-fn insert_data_positions(
+fn resolve_constant_expressions(
```

Comment:
I think the main purpose of this is still to substitute references to data items by their absolute positions in memory.